state_machine/drivers/pycubedmini/lib/pycubed.py

```diff
@@ -262,394 +306,168 @@ def reinit(self, dev):
-        create a new log file
-        """
-        if self.hardware['SDcard']:
-            n = 0
-
-            # iterate through all files in the logs folder
-            for f in listdir('/sd/logs/'):
-                # if the file number is greater than n, set n to file number
-                if int(f[3: -4]) > n:
-                    n = int(f[3: -4])
-
-            # the new log file has number n + 1; n is the current
-            # greatest file number
-            self.logfile = "/sd/logs/log" + "{:03}".format(n + 1) + ".txt"
-
-            # open the new logfile and write the time it was created +
-            # the current status
-            with open(self.logfile, "a") as log:
-                log.write('# Created: {:.0f}\r\n# Status: {}\r\n'.format(
-                    time.monotonic(), self.status))
-
-            # print a confirmation message that a new logfile was created
-            print('New log file:', self.logfile)
-
-    def print_file(self, filedir=None):
-        """
-        print a file given its directory; file directory is by default None
-        """
-
-        # if no file directory is passed, use the directory of the log file
-        if filedir is None:
-            filedir = self.logfile
-
-        print('--- Printing File: {} ---'.format(filedir))
-
-        # open the current file directory as read only, print line by line
-        # (removing whitespace)
-        with open(filedir, "r") as file:
-            for line in file:
-                print(line.strip())
-
-    def send_file(self, c_size, send_buffer, filename):
-        """
-        send a file given character size, buffer size, and the filename
-        """
-
-        # number of packets is the size of the filename / character size
-        num_packets = int(stat(filename)[6] / c_size)
-
-        # open the file
-        with open(filename, "rb") as f:
-            # for each packet
-            for i in range(num_packets + 1):
-                # move the cursor to the end of i * character size,
-                # add to buffer
-                f.seek(i * c_size)
-                f.readinto(send_buffer)
-
-                # return bytes; yield keyword returns without destroying
-                # states of local vars
-                yield bytes([i, 0x45, num_packets])
-
-    def save(self, dataset, savefile=None):
-        """
-        save the passed dataset to the passed savefile
-        dataset should be a set of lists; each line is a list:
-            save(([line1],[line2]))
-        to save a string, make it an item in a list:
-            save(['This is my string'])
-        by default, savefile is not passed
-        """
-        # if no savefile is passed, use the current filename attribute
-        # by default
-        if savefile is None:
-            savefile = self.filename
-
-        # open save file
+
+# initialize Satellite as cubesat
+cubesat = Satellite()
+
+"""
+IMU-related functions
+TODO: cubesat.imu hardware check
```

Comment:
Trying to remember the conclusion we came to on this. Did we decide that we just let the system error on `cubesat.imu` not existing?

Reply:
Yes, we were planning for the purposes of this pull request it should just let the system error on cubesat.imu. But later we'd like to to produce the correct error, that being some HardwareNotConnectedErrror or something

Reply:
Ok, that sounds good.

Since that is the case, and since there is now an issue for this, @yashikabatra15 please remove all of the lines that say `TODO: _____ hardware check`

Reply:
I'm implementing the error for the next PR, but yeah, for this one we're just going to let the system error. I'll remove all these lines